onnxruntime/contrib_ops/webgpu/bert/flash_attention.cc

```diff
@@ -140,15 +140,18 @@ Status FlashAttentionProgram::GenerateShaderCode(ShaderHelper& shader) const {
   const max_k_step: u32 = 16u;
   const vec_factor: u32 = 4u;
   const qkv_head_size_vec: u32 = qkv_head_size / vec_factor;
+  const half_qkv_head_size_vec = qkv_head_size_vec / 2u;
```

Comment:
we should add a constrait that qkv_head_size_vec  is divisble by 2. that is  parameters.head_size_ % 8 == 0; in line 496

Reply:
Nice catching. Done.